apiserver/common/firewall/mock_test.go

```diff
@@ -193,7 +192,10 @@ type mockStringsWatcher struct {
 
 func newMockStringsWatcher() *mockStringsWatcher {
 	w := &mockStringsWatcher{changes: make(chan []string, 1)}
-	go w.doneWhenDying()
+	w.Tomb.Go(func() error {
```

Comment:
We do this because doneWhenDying() selects from a tomb channel, right?

Reply:
Yes, with v1 the go func blocked until  Kill() was killed then called Done() and returned. With v2 Go() needs to be called otherwise Wait() will wait forever. So this does the same thing, a dummy func that just blocks until it gets a Kill() then returns, stopping the watcher.

We could do this with 2 channels, dying and dead and send those signals ourselves, but this does that for us.

It could also be simplified to:

```golang
w.Tomb.Go(func() error {
    <-w.Tomb.Dying()
    return nil
})
```